middlewares/osio/tenant.go

```diff
@@ -25,32 +39,56 @@ type attributes struct {
 }
 
 type namespace struct {
+	Name       string `json:"name"`
+	Type       string `json:"type"`
 	ClusterURL string `json:"cluster-url"`
 }
 
-func getClusterURL(resp response) (string, error) {
+func getNamesapce(resp response) (ns namespace, err error) {
 	if len(resp.Data.Attributes.Namespaces) == 0 {
-		return "", fmt.Errorf("unable to locate cluster url")
+		return ns, fmt.Errorf("unable to locate cluster url")
 	}
 
-	return resp.Data.Attributes.Namespaces[0].ClusterURL, nil
+	return resp.Data.Attributes.Namespaces[0], nil
 }
 
-func locateTenant(client *http.Client, tenantBaseURL, osioToken string) (string, error) {
+func locateTenant(client *http.Client, tenantBaseURL, token string) (ns namespace, err error) {
+	req, err := http.NewRequest("GET", tenantBaseURL+"/tenant", nil)
+	if err != nil {
+		return ns, err
+	}
+	req.Header.Set(Authorization, "Bearer "+token)
+	resp, err := client.Do(req)
+	if resp.StatusCode != http.StatusOK {
+		return ns, fmt.Errorf("Unknown status code " + resp.Status)
+	}
+	defer resp.Body.Close()
+
+	var r response
+	err = json.NewDecoder(resp.Body).Decode(&r)
+	if err != nil {
+		return ns, err
+	}
+	return getNamesapce(r)
+}
 
-	req, err := http.NewRequest("GET", tenantBaseURL+"/user/services", nil)
+func locateTenantByID(client *http.Client, tenantBaseURL, token, userID string) (ns namespace, err error) {
+	url := fmt.Sprintf("%s/tenants/%s", tenantBaseURL, userID)
```

Comment:
/api/tenants/<user-id> along with che_service_account_token used to get Cluster details.